File: 260/solution.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        tmp = 0
        for num in nums:
            tmp ^= num
        i = 1
        while i & tmp == 0:
            i <<= 1
        a, b = 0, 0
        for num in nums:
            if num & i:
                a ^= num
            else:
                b ^= num
        return [a, b]

File: 260/Solution.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int tmp = 0;
        for (int num : nums) {
            tmp ^= num;
        }
        int i = 1;
        while ((i & tmp) == 0) {
            i <<= 1;
        }
        int a = 0, b = 0;
        for (int num : nums) {
            if ((num & i) > 0) {
                a ^= num;
            } else {
                b ^= num;
            }
        }
        return new int[] { a, b };
    }
}